nilearn/surface/surface.py

```diff
@@ -442,6 +475,16 @@ def vol_to_surf(img, surf_mesh,
         Samples falling out of this mask or out of the image are ignored.
         If ``None``, don't apply any mask.
 
+    inner_mesh : str or numpy.ndarray, optional (default=None)
+        Either a file containing a surface mesh or a pair of ndarrays
+        (coordinates, triangles). If provided this is an inner surface that is
+        nested inside the one represented by `surf_mesh` -- e.g. `surf_mesh` is
+        a pial surface and `inner_mesh` a white matter surface. In this case
+        nodes in both meshes must correspond: node i in `surf_mesh` is just
+        across the gray matter thickness from node i in `inner_mesh`. Image
+        values for index i are then sampled along the line joining these two
+        points.
+
```

Comment:
My understanding is that quite often, people want to select the depth at which they're sampling (layered fMRI). This only makes sense for high resolution data of course. 
So I would define the sampling scheme along the cortical depths by a list of values between 0 and 1: [0]=white, [1]=pial, [.5]=middle surface, [0, 0.1,..,1]= average across 10 intermediate surfaces.

Reply:
ok! should that be an extra parameter then? or we can deprecate the "ball" sampling strategy and apply this depth selection either to the vertex normal or to the between-mesh segment, depending on whether a second surface is provided or not. In the single mesh /fixed depth case users may also want to select the offsets of sample points. for now they can change the scale but the points are always centered on the vertex

Reply:
I think that the ball think is a very good idea if you have only one surface. Ideally, I would like to make it possible to use both heuristics. It is also better for backward compatibility.
The problem is how to have both in the same API, and making it clear !